demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,508 @@
+        std::ifstream inputLabelsFile(FLAGS_labels);
+        std::string labelsLine;
+        while (std::getline(inputLabelsFile, labelsLine)) {
+            labels.push_back(labelsLine.substr(0, labelsLine.find(',')));
+        }
+        // ---------------------------------------------------------------------------------------------------
+
+        // -------------------------------------------Read network--------------------------------------------
+        Core ie;
+        CNNNetwork network = ie.ReadNetwork(FLAGS_m);
+        // ---------------------------------------------------------------------------------------------------
+
+        // ------------------------------------------Reshape network------------------------------------------
+        auto input_shapes = network.getInputShapes();
+        std::string input_name;
+        SizeVector input_shape;
+        std::tie(input_name, input_shape) = *input_shapes.begin();
+        int modelInputResolution = input_shapes[input_name][2];
+        input_shape[0] = FLAGS_b;
+        input_shape[2] = input_shapes[input_name][2];
+        input_shape[3] = input_shapes[input_name][3];
+        input_shapes[input_name] = input_shape;
+        std::cout << "Resizing network to the image size = [" 
+                  << input_shapes[input_name][2] << "x" << input_shapes[input_name][3]
+                  << "] " << "with batch = " << FLAGS_b << std::endl;
+        network.reshape(input_shapes);
+        // ---------------------------------------------------------------------------------------------------
+
+        // ----------------------------------------Init inputBlobName-----------------------------------------
+        InputsDataMap inputInfo(network.getInputsInfo());
+        std::string inputBlobName = inputInfo.begin()->first;
+        // ---------------------------------------------------------------------------------------------------
+
+        // -----------------------------------------Configure layers------------------------------------------
+        std::map<std::string, std::vector<unsigned long>> inputBlobsDimsInfo;
+        std::map<std::string, std::vector<unsigned long>> outputBlobsDimsInfo;
+        for (auto inputBlobsIt = inputInfo.begin(); inputBlobsIt != inputInfo.end(); ++inputBlobsIt) {
+            auto layerName = inputBlobsIt->first;
+            auto layerData = inputBlobsIt->second;
+            auto layerDataDims = layerData->getTensorDesc().getDims();
+
+            std::vector<unsigned long> layerDims(layerDataDims.data(), layerDataDims.data() + layerDataDims.size());
+            inputBlobsDimsInfo[layerName] = layerDims;
+
+            if (layerDataDims.size() == 4) {
+                layerData->setLayout(Layout::NCHW);
+                layerData->setPrecision(Precision::U8);
+            } else if (layerDataDims.size() == 2) {
+                layerData->setLayout(Layout::NC);
+                layerData->setPrecision(Precision::FP32);
```

Comment:
It sets data to be `float`, but later the line `matU8ToBlob<uint8_t>(emptyIR.images[i].mat, inputBlob, i);` treats the data as `uint8_t` unconditionally. This option can be removed if there are no models in that format in OMZ.